### ISSUE TITLE
Using configuration APIs in a CLI tool cannot read content files

### ISSUE THREAD

**Issue by johnkors** [NuGet/Home#5302]
## Details about Problem
When creating a dotnet core tool (CLI extension), I'd like for the tools to include a content file (appsettings.json), so I can use configuration APIs when running the tool. 

When installing the tool from Nuget, the content file cannot be found using the configuration APIs because it's trying to locate it in the `.tools` folder instead of inside the nuget package folder:

I'm using this regular config building pattern: 

```
            var builder = new ConfigurationBuilder().AddJsonFile("appsettings.json");
            Configuration = builder.Build();
```

Csproj:

```
    <PackageType>DotnetCliTool</PackageType>
    <Content Include="appsettings.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
      <PackagePath>lib\netcoreapp1.0</PackagePath>
    </Content>
```

Runtime, the tool tries to find the file here:

`<HOME>\.nuget\packages\.tools\dotnet-retire\1.0.3-beta008\netcoreapp1.0\appsettings.json`

when (imo) it should be trying to find it here:
`<HOME>\.nuget\packages\dotnet-retire\1.0.3-beta008\lib\netcoreapp1.0`\appsettings.json
(I've verified the file exists as content in this folder)

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe):
```
λ dotnet --info
.NET Command Line Tools (1.0.4)

Product Information:
 Version:            1.0.4
 Commit SHA-1 hash:  af1e6684fd

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.3.9600
 OS Platform: Windows
 RID:         win81-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.4
```

NuGet version (x.x.x.xxx):

dotnet.exe --version (if appropriate):
```
λ dotnet --version
1.0.4
```
VS version (if appropriate):

OS version (i.e. win10 v1607 (14393.321)):
`win81-x64`

## Detailed repro steps so we can see the same problem

1. Create a dotnet cli tool and include a `appsettings.json` file as content somewhere in the NuGet. Make the tool try to read that file using the configuration APIs.

2. Install the tool in a host project via   `<DotNetCliToolReference Include="mytool" Version="1.0.0-*" />`

3. Run the tool via `dotnet <mytool>` => Exception, because it tries to find the content in the wrong folder

...

### Sample Project
https://github.com/RetireNet/dotnet-retire/commit/07dc0e9f3c346a7bc0f15485f5f7164e740b581a

**Comment by mishra14** [comment]
Due to dotnet cli's move towards global tools in cli 2.1.0, it seems unlikely that we are not going to get to this.

@johnkors please let us know if you are still blocked by this.

**Comment by johnkors** [comment]
Do you have proposal for how to work with configuration APIs in the context of a CLI tool then? At the moment it's broken, and I ended up hardcoding constants instead.

**Comment by nkolev92** [comment]
Lower priority scenario.

**Comment by johnkors** [comment]
What's the recommended location for config files for global tools?
